core/src/SYCL/Kokkos_SYCL_Parallel_Range.hpp

```diff
@@ -71,7 +71,7 @@ class Kokkos::Impl::ParallelFor<FunctorType, ExecPolicy,
         *space.impl_internal_space_instance();
     cl::sycl::queue& q = *instance.m_queue;
 
-    q.wait();
+    q.wait_and_throw();
```

Comment:
@dhollman says:
> (Here and below) couldn't/shouldn't this just be space.fence()?

Reply:
Addressed in [`40c6dbc` (#3609)](https://github.com/kokkos/kokkos/pull/3609/commits/40c6dbc90c4ee515c193cfd1680ad8c93ac40afc).